infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/jwtproxy/JwtProxyConfigBuilder.java

```diff
@@ -37,46 +51,56 @@ public void addVerifierProxy(Integer listenPort, String upstream, Set<String> ex
-                  + "      upstream: %s/\n" // http://localhost:4401
-                  + "      audience: %s\n"
-                  + "      max_skew: 1m\n"
-                  + "      max_ttl: 8800h\n"
-                  + "      key_server:\n"
-                  + "        type: preshared\n"
-                  + "        options:\n"
-                  + "          issuer: wsmaster\n"
-                  + "          key_id: %s\n"
-                  + "          public_key_path: "
-                  + JWT_PROXY_CONFIG_FOLDER
-                  + "/"
-                  + JWT_PROXY_PUBLIC_KEY_FILE
-                  + "\n"
-                  + "      claims_verifiers:\n"
-                  + "      - type: static\n"
-                  + "        options:\n"
-                  + "          iss: wsmaster\n"
-                  + "      nonce_storage:\n"
-                  + "        type: void\n",
-              verifierProxy.listenPort,
-              verifierProxy.upstream,
-              workspaceId,
-              workspaceId));
+      VerifierConfig verifierConfig =
+          new VerifierConfig()
+              .withAudience(workspaceId)
+              .withUpstream(verifierProxy.upstream)
+              .withMaxSkew("1m")
+              .withMaxTtl("8800h")
+              .withKeyServer(
+                  new RegistrableComponentConfig()
+                      .withType("preshared")
+                      .withOptions(
+                          ImmutableMap.of(
+                              "issuer",
+                              "wsmaster",
+                              "key_id",
+                              workspaceId,
+                              "public_key_path",
+                              JWT_PROXY_CONFIG_FOLDER + '/' + JWT_PROXY_PUBLIC_KEY_FILE)))
+              .withClaimsVerifier(
+                  Collections.singleton(
+                      new RegistrableComponentConfig()
+                          .withType("static")
+                          .withOptions(ImmutableMap.of("iss", "wsmaster"))))
+              .withNonceStorage(new RegistrableComponentConfig().withType("void"));
+
       if (!verifierProxy.excludes.isEmpty()) {
-        configBuilder.append("      excludes:\n");
-        verifierProxy.excludes.forEach(s -> configBuilder.append(format("      - %s\n", s)));
+        verifierConfig.setExcludes(verifierProxy.excludes);
       }
+
+      VerifierProxyConfig proxyConfig =
+          new VerifierProxyConfig()
+              .withListenAddr(":" + verifierProxy.listenPort)
+              .withVerifierConfig(verifierConfig);
+
+      proxyConfigs.add(proxyConfig);
     }
 
-    configBuilder.append("  signer_proxy:\n" + "    enabled: false\n");
-    return configBuilder.toString();
+    try {
+      return YAML_PARSER.writeValueAsString(config);
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Error during creation of JWTProxy config YAML: ", e);
```

Comment:
It would be better to use  `InfrastructureException` or even `InternalInfrastructureException` instead of `RuntimeException` because I'm not sure that `RuntimeException` will be handled correctly.
Also, I guess `"Error during creation of JWTProxy config YAML: "` is incomplete and `e.getMessage` should be added there.